TestFoundation/TestCalendar.swift

```diff
@@ -223,4 +224,16 @@ class TestNSDateComponents: XCTestCase {
         XCTAssertEqual(components.hour, 12)
         XCTAssertEqual(copy.hour, 14)
     }
+
+    func test_dateDifferenceComponents() {
+        let date1 = Date(timeIntervalSince1970: 0)
+
+        // 1971-06-21
+        let date2 = Date(timeIntervalSince1970:  46310400)
+        let difference = Calendar.current.dateComponents([.month, .year, .day], from: date1, to: date2)
```

Comment:
for leap months you probably need the Chinese calendar iirc.

Reply:
On Darwin we actually almost always add "is leap month" to `DateComponents` in output, if that's what you're referring to. (Leap months are also applicable in the Hebrew calendar, and otherwise.)

Reply:
(But yes, in this case, on Darwin we don't add "is leap month" here.)